### PR TITLE
Start implementing features for BasicTest.vm

### PR DESCRIPTION
**Pull request by wildmaples**
We're getting BasicTest.vm to work. It's not done yet, more changes in the next PR.

## Changes

* Implement pop command for segments local and arguments

### CODE REVIEW

File: lib/parser.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ def command_type
   end
 
   def arg1
-    if command_type == :C_PUSH
+    if command_type == :C_PUSH || command_type == :C_POP
```
Comment (tomstuart) [inline lib/parser.rb:32]: Now that we’re comparing a single variable against multiple values, another option is something like:

```suggestion
    if [:C_PUSH, :C_POP].include?(command_type)
```

Or we could turn it into a `case command_type` and say `when :C_PUSH, :C_POP`. So many choices!

File: lib/code_writer.rb [hunk 2]
```diff
@@ -27,7 +49,7 @@ def write_arithmetic(command)
     @out.puts <<~EOF
       @SP
     EOF
-
```
Comment (tomstuart) [inline lib/code_writer.rb:30]: 👨🏻‍🍳👌🏻

File: test/integration/vm_translator_test.rb [hunk 3]
```diff
@@ -24,4 +24,10 @@ def test_integration_test_stack_test
     expected = File.read("test/fixtures/StackTest.asm")
     assert_equal(expected, assembly_code)
   end
+
+  def test_integration_test_basic_test
+    assembly_code = `bin/vm-translator examples/BasicTest.vm`
+    expected = File.read("test/fixtures/BasicTest.asm")
+    assert_equal(expected, assembly_code)
+  end
```
Comment (tomstuart) [inline test/integration/vm_translator_test.rb:32]: We don’t need to fix it now, but note that this file is getting extremely repetitive: all of the tests are identical except for the `Foo.vm`/`Foo.asm` filenames. At some point we could clean this up with [metaprogramming](https://lpil.uk/blog/drying-up-minitest-tests-with-metaprogramming/) by writing a loop that calls `define_method` to dynamically define a test for each file we want to run through `bin/vm-translator`. (We could hardcode those filenames, but we could also determine them at runtime by [listing the contents](https://rubydoc.info/stdlib/core/Dir.glob) of the `examples/` directory.)

**Review by tomstuart — APPROVED** [review]
👏🏻

File: test/integration/vm_translator_test.rb (hunk 3 above)
Comment (wildmaples) [reply test/integration/vm_translator_test.rb:32]: Good idea! I created an issue so we remember this - https://github.com/wildmaples/jack-virtual-machine/issues/8

**Pull request merged**